vocab-builder-core/src/main/java/com/github/tkurz/sesame/vocab/VocabBuilder.java

```diff
@@ -190,73 +197,150 @@ public void generate(String className, PrintWriter out) throws IOException, Grap
-            out.println();
-        }
-
-        //static init
-        out.println(getIndent(1) + "static {");
-        out.printf(getIndent(2) + "ValueFactory factory = ValueFactoryImpl.getInstance();%n");
-        out.println();
-        for (String key : keys) {
-            String nextKey = cleanKey(doCaseFormatting(key, getConstantCase()));
-            out.printf(getIndent(2) + "%s = factory.createURI(%s.NAMESPACE, \"%s\");%n", nextKey, className, key);
         }
-        out.println(getIndent(1) + "}");
-        out.println();
+		// do the entire uri constant generation
+		if ( uriGeneration != null ) {
+	        for (String key : keys) {
+	            Literal comment = getFirstExistingObjectLiteral(model, splitUris.get(key), getPreferredLanguage(), COMMENT_PROPERTIES);
+	            Literal label = getFirstExistingObjectLiteral(model, splitUris.get(key), getPreferredLanguage(), LABEL_PROPERTIES);
+
+	            out.println(getIndent(1) + "/**");
+	            if (label != null) {
+	                out.printf(getIndent(1) + " * %s%n", label.getLabel());
+	                out.println(getIndent(1) + " * <p>");
+	            }
+	            out.printf(getIndent(1) + " * {@code %s}.%n", splitUris.get(key).stringValue());
+	            if (comment != null) {
+	                out.println(getIndent(1) + " * <p>");
+	                out.printf(getIndent(1) + " * %s%n", WordUtils.wrap(comment.getLabel().replaceAll("\\s+", " "), 70, "\n" + getIndent(1) + " * ", false));
+	            }
+	            out.println(getIndent(1) + " *");
+	            out.printf(getIndent(1) + " * @see <a href=\"%s\">%s</a>%n", splitUris.get(key), key);
+	            out.println(getIndent(1) + " */");
+				final String nextKey = cleanKey(
+						String.format("%s%s%s", StringUtils.defaultString(uriGeneration.getConstantPrefix()),
+								doCaseFormatting(key, uriGeneration.getCaseFormat()),
+								StringUtils.defaultString(uriGeneration.getConstantSuffix())));
+
+	            //String nextKey = cleanKey(doCaseFormatting(key, uriGeneration.getCaseFormat()));
+	            checkField(className, nextKey);
+	            out.printf(getIndent(1) + "public static final URI %s;%n", nextKey);
+	            out.println();
+	        }
+	        //static init
+	        out.println(getIndent(1) + "static {");
+	        out.printf(getIndent(2) + "ValueFactory factory = ValueFactoryImpl.getInstance();%n");
+	        out.println();
+	        for (String key : keys) {
+				final String nextKey = cleanKey(
+						String.format("%s%s%s", StringUtils.defaultString(uriGeneration.getConstantPrefix()),
+								doCaseFormatting(key, uriGeneration.getCaseFormat()),
+								StringUtils.defaultString(uriGeneration.getConstantSuffix())));
+	            out.printf(getIndent(2) + "%s = factory.createURI(%s.NAMESPACE, \"%s\");%n", nextKey, className, key);
+	        }
+	        out.println(getIndent(1) + "}");
+	        out.println();
+
+		}
+        //string constant values
```

Comment:
Delete the following commented out code, and after reformatting the code above to use spaces, git will show what the changes you made to this section are.